app/actions/remote/post.ts

```diff
@@ -552,3 +552,38 @@ export const togglePinPost = async (serverUrl: string, postId: string) => {
         return {error};
     }
 };
+
+export const markPostAsUnread = async (serverUrl: string, postId: string, channelId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+    try {
+        const userId = await queryCurrentUserId(database);
+        const myChannel = await queryMyChannel(database, channelId);
+
+        if (myChannel) {
+            client.markPostAsUnread(userId, postId);
+            const lastViewed = Date.now();
+            await database.write(async () => {
+                await myChannel.update((m) => {
```

Comment:
Thanks @larkox .  I would need some help here with regards to the messageCount.  This is what I get from the server after making the API call - while marking one post on the Reception channel as unread. 

```
 {
		"channel_id": "zw43c5ttrjyu9dg7jnudwuz6bw",
		"last_viewed_at": 1635794782966, // time in milliseconds when the user last viewed the channel.
		"mention_count": 0,
		"mention_count_root": 0,
		"msg_count": 6610,
		"msg_count_root": 7925,
		"team_id": "rcgiyftm7jyrxnma1osd8zswby",
		"user_id": "9ciscaqbrpd6d8s68k76xb9bte"
}
```

However, the values for that record in the MyChannel table is 

```
"mentions_count": 0,
"message_count": 0,
```
I'm not sure how to drop down the message count.

Reply:
Worst case scenario, you can fetch the channel from the server, and that model will have the "TotalMsgCount" to compare with the msg_count received from the server. That being said, not sure if @enahum knows if we are storing that information somewhere in the database.

Reply:
ok so this is how it goes.. 

When you mark a post as unread, the server is not returning the mention or message count. But since you are marking the post as unread in the client it can be inferred that you have all the posts that are after this one, which then from there you can get from the local data which ones have mentioned you + if there are any other unread posts.

This means that in the local action (we are going to move this to a local action as @larkox suggested) we can then get those posts (excluding the current user own posts) that will give you the total of unreads, then iterate through them to see if there are any mentions (this needs to check all mention keys) and then you can get the total mention count.

With the above calculated data then you set `manuallyUnread: true`, `isUnread:true` , `lastViewedAt: post.create_at`, `viewedAt: post.create_at` and `mentionCount: <the calculated mentions value>`, `messageCount: <the calculated unread value>`

I believe this is how it was done previously and how it is done in the webapp.. perfect? not by any means, but this was done to save the server computation cycles as we had the data locally. (@hmhealey can you confirm this?)

@anurag6713 not sure how is this going to affect you.. perhaps we can identify what needs to happen either marking the channel as unread or a thread based on a flag added as an argument to this function.

Reply:
To simplify that process, we could just do:
```
await client.markPostAsUnread...
const [channel, member] = await Promise.All(client.getChannel, client.getMyChannelMember....)
and then use the mentions count from the response and unreads = channel.total_msg_count - member.msg_count
```

now let's chat about pros & cons of both solutions

Reply:
After some analysis, my previous logic was faulty, I missed the cases for mentions that follow the `Reply notifications` settings and the channel specific settings for `ignore system wide mentions` + the LDAP groups stuff + anything new that may set a mention for some reason. So in order to have always one source of truth in this case I think we need to rely on the server and fetch that information.